sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SocketFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.alibaba.csp.sentinel.transport.heartbeat.client;
+
+import java.io.IOException;
+import java.net.Socket;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import com.alibaba.csp.sentinel.transport.endpoint.Protocol;
+import com.alibaba.csp.sentinel.transport.ssl.SslFactory;
+
+/**
+ * @author Leo Li
+ */
+public class SocketFactory {
+
+    private static class SSLSocketFactoryInstance {
+        private static final SSLSocketFactory SSL_SOCKET_FACTORY = SslFactory.getSslConnectionSocketFactory().getSocketFactory();
+    }
+
+    public static Socket getSocket(Protocol protocol) throws IOException {
+        return protocol == Protocol.HTTP ? new Socket() : getSslSocketFactory().createSocket();
```

Comment:
What's the difference between the wrapped form and `Class.varible` form?

Reply:
I just wrote a small piece testing code and didn't find any different.

Reply:
> What's the difference between the wrapped form and `Class.varible` form?

This delays init, and threads are safe. Originally written directly in the static block of the class, now it is changed to the inner class.

Reply:
> 
> This delays init, and threads are safe. Originally written directly in the static block of the class, now it is changed to the inner class.

Yeah, I know. I'm just curious whether it's different to use `SSLSocketFactoryInstance.SSL_SOCKET_FACTORY` simply instead of `getSslSocketFactory()`?
My testing code is for these two approaches.

Reply:
> > This delays init, and threads are safe. Originally written directly in the static block of the class, now it is changed to the inner class.
> 
> Yeah, I know. I'm just curious whether it's different to use `SSLSocketFactoryInstance.SSL_SOCKET_FACTORY` simply instead of `getSslSocketFactory()`?
> My testing code is for these two approaches.

Oh, it doesn't make any difference, it's a "copy" issue:smile: